### PR TITLE
Usage and Role Operation Additions

### LINKED ISSUE #163: Add new Usage Operations

### PR DESCRIPTION
**Pull request by paulsmelser**
resolves #163

### CODE REVIEW

File: src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java [hunk 1]
```diff
@@ -0,0 +1,15 @@
+package org.springframework.social.partnercenter.api.billing.usage;
+
+import java.math.BigDecimal;
+
+public class SpendingBudget {
```
Comment (ian-beckles) [inline src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: Is this related to Budget? i.e. Should this extend Budget or should Budget be an Interface that this implements.

Comment (paulsmelser) [reply src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: Your right. I saw the SpendingBudget object in the microsoft SDK but I now see I had implemented it as Budget. SpendingBudget is redundant. I'll modify the Budget class though to use BigDecimal and extend ResourceBase however to be in keeping with more recent changes

File: src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java [hunk 2]
```diff
@@ -2,14 +2,17 @@
...
-public class SpendingBudget {
-	private BigDecimal amount;
+import org.springframework.social.partnercenter.api.ResourceBase;
 
-	public BigDecimal getAmount() {
-		return amount;
+public class SpendingBudget extends ResourceBase{
+	private BigDecimal ammount;
+
+	public BigDecimal getAmmount() {
```
Comment (ian-beckles) [inline src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: getAmount

File: src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java [hunk 3]
```diff
@@ -2,14 +2,17 @@
...
+		return ammount;
 	}
 
-	public void setAmount(BigDecimal amount) {
-		this.amount = amount;
+	public SpendingBudget setAmmount(BigDecimal ammount) {
```
Comment (ian-beckles) [inline src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: setAmount

File: src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java [hunk 4]
```diff
@@ -2,14 +2,17 @@
 
 import java.math.BigDecimal;
 
-public class SpendingBudget {
-	private BigDecimal amount;
+import org.springframework.social.partnercenter.api.ResourceBase;
 
-	public BigDecimal getAmount() {
-		return amount;
+public class SpendingBudget extends ResourceBase{
```
Comment (ian-beckles) [inline src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: Space after ResourceBase

File: src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java (hunk 2 above)
Comment (paulsmelser) [reply src/main/java/org/springframework/social/partnercenter/api/billing/usage/SpendingBudget.java]: ha
awesome
